src/geometry.nim

```diff
@@ -274,6 +274,26 @@ proc det*[N: static[int], V](m: SQMat[N, V]): V =
         quit "Determinant is only implemented for matrices at most 4x4."
 
 
+proc solve*(mat: Mat3f, vec: Vec3f): Vec3f {.raises: ValueError.} =
+    let det = mat.det
+    if det == 0.0: raise newException(ValueError, "Matrix is not invertible.")
+    
+    # Create matrices for each variable by replacing the corresponding column
+    var matX = mat
```

Comment:
Here i propose a minor change: instead of having
```nim
    var matX = mat
    var matY = mat
    var matZ = mat
```
we could just have
```nim
var
    matX = mat
    matY = mat
    matZ = mat
```